types/trace/trace.go

```diff
@@ -3,6 +3,7 @@ package trace
 
```

Comment:
Maybe this file deserves a split since it's already huge.

Reply:
You are right. I started doing it in this PR and it go way too big.
Let's leave it for another PR.